pkg/machine/ignition/ready.go

```diff
@@ -0,0 +1,36 @@
+package ignition
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/containers/podman/v4/pkg/machine/define"
+)
+
+type ReadyUnitOpts struct {
+	Port uint64
+}
+
+func CreateReadyUnitFile(provider define.VMType, opts *ReadyUnitOpts) (string, error) {
+	readyUnit := DefaultReadyUnitFile()
+	switch provider {
+	case define.QemuVirt:
+		readyUnit.Add("Unit", "Requires", "dev-virtio\\x2dports-vport1p1.device")
+		readyUnit.Add("Unit", "After", "systemd-user-sessions.service")
+		readyUnit.Add("Service", "ExecStart", "/bin/sh -c '/usr/bin/echo Ready >/dev/vport1p1'")
+	case define.AppleHvVirt:
+		readyUnit.Add("Unit", "Requires", "dev-virtio\\x2dports-vsock.device")
+		readyUnit.Add("Service", "ExecStart", "/bin/sh -c '/usr/bin/echo Ready | socat - VSOCK-CONNECT:2:1025'")
+	case define.HyperVVirt:
+		if opts == nil {
```

Comment:
To be future proof, I think that `Port` needs to be a pointer (and cheked here) as well.  Otherwise, when adding a new option, the check won't hit anymore and we'd use Port 0.

Reply:
Or check for `Port != 0`.

Reply:
done